Assets/MixedRealityToolkit/Inspectors/MixedRealityToolkitFacadeHandler.cs

```diff
@@ -0,0 +1,163 @@
+            EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+        }
+
+        private static void OnPlayModeStateChanged(PlayModeStateChange obj)
+        {
+            if (!MixedRealityToolkit.IsInitialized)
+            {
+                return;
+            }
+
+            // When the play state changes just nuke everything and start over
+            DestroyAllChildren();
+        }
+
+        private static void UpdateServiceFacades(SceneView sceneView)
+        {
+            // If we're not using inspectors, destroy them all now
+            if (!MixedRealityToolkit.IsInitialized || MixedRealityToolkit.Instance.HasActiveProfile && !MixedRealityToolkit.Instance.ActiveProfile.UseServiceInspectors)
+            {
+                DestroyAllChildren();
+                return;
+            }
+
+            servicesToSort.Clear();
+
+            foreach (IMixedRealityService service in MixedRealityToolkit.Instance.ActiveSystems.Values)
+            {
+                servicesToSort.Add(service);
+            }
+
+            foreach (Tuple<Type, IMixedRealityService> registeredService in MixedRealityToolkit.Instance.RegisteredMixedRealityServices)
+            {
+                servicesToSort.Add(registeredService.Item2);
+            }
+
+            servicesToSort.Sort(
+                delegate (IMixedRealityService s1, IMixedRealityService s2)
+                {
+                    string s1Name = s1.GetType().Name;
+                    string s2Name = s2.GetType().Name;
+
+                    if (s1Name == s2Name)
+                    {
+                        return s1.Priority.CompareTo(s2.Priority);
+                    }
+
+                    return s1Name.CompareTo(s2Name);
+                });
+
+            for (int i = 0; i < servicesToSort.Count; i++)
+            {
+                CreateFacade(MixedRealityToolkit.Instance.transform, servicesToSort[i], i);
+            }
+
+            // Delete any stragglers
+            childrenToDelete.Clear();
+            for (int i = servicesToSort.Count; i < MixedRealityToolkit.Instance.transform.childCount; i++)
+            {
+                childrenToDelete.Add(MixedRealityToolkit.Instance.transform.GetChild(i));
+            }
+
+            foreach (Transform childToDelete in childrenToDelete)
+            {
+                if (Application.isPlaying)
+                {
+                    GameObject.Destroy(childToDelete.gameObject);
+                }
+                else
+                {
+                    GameObject.DestroyImmediate(childToDelete.gameObject);
+                }
+            }
+
+            // Update all self-registered facades
+            foreach (ServiceFacade facade in ServiceFacade.ActiveFacadeObjects)
+            {
+                if (facade == null)
+                {
+                    continue;
+                }
+
+                facade.CheckIfStillValid();
+            }
```

Comment:
There's a bug here where an exception is thrown if you:

1) Enable facades
2) Try to drag any of the facades into like MixedRealityInputSystem into MixedRealityPlayspace.
Hit an exception:

System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.ThrowHelper.ThrowInvalidOperationException (System.ExceptionResource resource) [0x0000b] in <d7ac571ca2d04b2f981d0d886fa067cf>:0 
  at System.Collections.Generic.List`1+Enumerator[T].MoveNextRare () [0x00013] in <d7ac571ca2d04b2f981d0d886fa067cf>:0 
  at System.Collections.Generic.List`1+Enumerator[T].MoveNext () [0x0004a] in <d7ac571ca2d04b2f981d0d886fa067cf>:0 
  at Microsoft.MixedReality.Toolkit.Utilities.Facades.MixedRealityToolkitFacadeHandler.UpdateServiceFacades (UnityEditor.SceneView sceneView) [0x0021e] in D:\github\MixedRealityToolkit-Unity\Assets\MixedRealityToolkit\Inspectors\MixedRealityToolkitFacadeHandler.cs:97 
  at (wrapper delegate-invoke) <Module>.invoke_void_SceneView(UnityEditor.SceneView)
  at UnityEditor.SceneView.CallOnSceneGUI () [0x000da] in C:\buildslave\unity\build\Editor\Mono\SceneView\SceneView.cs:2848 
  at UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () [0x0000d] in C:\buildslave\unity\build\Editor\Mono\SceneView\SceneView.cs:2125 
  at UnityEditor.SceneView.OnGUI () [0x00376] in C:\buildslave\unity\build\Editor\Mono\SceneView\SceneView.cs:1986 
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <d7ac571ca2d04b2f981d0d886fa067cf>:0  occurred

The issue is that face.CheckIfStillValid will destroy the facade (i.e. it will destroy the input system one and then recreate), but this changes the ActiveFacadeObjects collection (because in the ServiceFacade's own  OnDestroy method, it removes itself from the ActiveFacadeObjects list).

The work is probably then to snap a copy of the this array and iterate over the copy of the array instead.